vmaas/reposcan/reposcan.py

```diff
@@ -180,6 +183,58 @@ class SyncHandler:
 
     task_type = "Unknown"
 
+    @staticmethod
+    def fetch_git():
+        """Download and parse data from configured git repo"""
+        if IS_FEDRAMP:
+            LOGGER.info("FedRAMP env, using static assets source: %s", REPOLIST_STATIC_DIR)
+            repolist_dir = REPOLIST_STATIC_DIR
+        else:
+            LOGGER.info("Downloading assets from git %s", REPOLIST_GIT)
+            shutil.rmtree(REPOLIST_DIR, True)
+            os.makedirs(REPOLIST_DIR, exist_ok=True)
+
+            # Should we just use replacement or add a url handling library, which
+            # would be used replace the username in the provided URL ?
+            git_url = REPOLIST_GIT.replace('https://', f'https://{REPOLIST_GIT_TOKEN}:x-oauth-basic@')
+            git_ref = REPOLIST_GIT_REF if REPOLIST_GIT_REF else 'master'
+
+            git.Repo.clone_from(git_url, REPOLIST_DIR, branch=git_ref)
+            repolist_dir = REPOLIST_DIR
+
+        if not os.path.isdir(repolist_dir):
+            LOGGER.error("Downloading assets git repo failed: Directory was not created")
+            return None, None, None
+
+        products, repos = {}, []
+
+        paths = REPOLIST_PATH.split(',')
+        for path in paths:
+            # Trim the spaces so we can have nicely formatted comma lists
+            path = path.strip()
+            if not os.path.isfile(repolist_dir + '/' + path):
+                LOGGER.error("Downloading git repo failed: File %s was not created", path)
+                return None, None, None
+
+            with open(repolist_dir + '/' + path, 'r', encoding='utf8') as json_file:
+                data = json.load(json_file)
+            assert data
```

Comment:
do you want to raise an exception when data is empty or why is this needed?

Reply:
removed assert, if data is empty is checked in function just below this anyway